### PR TITLE
Remove unnecessary env variables passing

### PR DESCRIPTION
**Pull request by goganchic**
PR into docs: https://github.com/getsentry/develop/pull/628

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -264,15 +264,15 @@ services:
       context: ./cron
       args:
         BASE_IMAGE: "$SNUBA_IMAGE"
-    command: '"*/5 * * * * gosu snuba snuba cleanup --storage errors --dry-run False"'
+    command: '"*/5 * * * * /usr/sbin/gosu snuba snuba cleanup --storage errors --dry-run False"'
```
Comment (aminvakil) [inline docker-compose.yml:267]: This is not related to this PR and we're not still sure if there is a problem at all, as no one else has reported this issue until now. Let's keep the discussion in https://github.com/getsentry/self-hosted/issues/1556.

File: install/build-docker-images.sh [hunk 2]
```diff
@@ -3,8 +3,8 @@ echo "${_group}Building and tagging Docker images ..."
 echo ""
 # Build any service that provides the image sentry-self-hosted-local first,
 # as it is used as the base image for sentry-cleanup-self-hosted-local.
-$dc build --build-arg "http_proxy=${http_proxy:-}" --build-arg "https_proxy=${https_proxy:-}" --build-arg "no_proxy=${no_proxy:-}" --force-rm web
-$dc build --build-arg "http_proxy=${http_proxy:-}" --build-arg "https_proxy=${https_proxy:-}" --build-arg "no_proxy=${no_proxy:-}" --force-rm
```
Comment (aminvakil) [inline install/build-docker-images.sh:7]: Doesn't this break current users' workflow using `http_proxy` env variable?

File: install/install-wal2json.sh [hunk 3]
```diff
@@ -5,9 +5,7 @@ ARCH=$(uname -m)
 FILE_NAME="wal2json-Linux-$ARCH-glibc.so"
 
 docker_curl() {
-    # The environment variables can be specified in lower case or upper case.
-    # The lower case version has precedence. http_proxy is an exception as it is only available in lower case.
-    docker run --rm -e http_proxy -e https_proxy -e HTTPS_PROXY -e no_proxy -e NO_PROXY curlimages/curl:7.77.0 "$@"
+    docker run --rm curlimages/curl:7.77.0 "$@"
```
Comment (aminvakil) [inline install/install-wal2json.sh:8]: Also here.

**Review by aminvakil — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (goganchic) [reply docker-compose.yml]: My mistake, yep. Removed this commit from PR.

File: install/build-docker-images.sh (hunk 2 above)
Comment (goganchic) [reply install/build-docker-images.sh:7]: Yes you are right. But I think it's very specific scenario: such users can access the Dockerhub without any proxy, but need http proxy for some resources during Sentry installation, like https://github.com/getsentry/wal2json.

For users who access any Internet resources (including Dockerhub) during installation it's required to set HTTP Proxy several times:
1. Docker systemd file for `docker pull`
2. `~/.docker/config.json` for any commands inside docker containers
3. `/etc/environment` for settings shell proxy variables

I think we can set only do 1. and 2. But in this case `docker build` will fail because in the commands above we explicitly overwrite proxy variables with empty values, that is why I suggest to modify these lines.

Actually we can keep these lines unchanged but from our discussion https://github.com/getsentry/develop/pull/622#discussion_r912366290 I'd found that this behavior is a bit unclear, that is why I'd created this PR with such changes.

Comment (goganchic) [reply install/build-docker-images.sh:7]: Let me check it one more time. I'd tried to install Sentry with my changes again and there are some issues.

Comment (goganchic) [reply install/build-docker-images.sh:7]: I'd rechecked this one more time. Proxies from `~/.docker/config.json` work only for `docker run` command and do not work for `docker build`. It is written in the [docker docs](https://docs.docker.com/network/proxy/):

> When you create or start new containers, the environment variables are set automatically within the container.

So we have to pass proxy env variables explicitly for the build command, that is why this PR shouldn't be merged.

**Comment by aminvakil** [comment]
@goganchic Thank you very much for your effort and getting to the bottom of this!
